x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/migrate_legacy_actions.ts

```diff
@@ -103,11 +113,14 @@ export const migrateLegacyActions: MigrateLegacyActions = async (
     context.logger.error(
       `migrateLegacyActions(): Failed to migrate legacy actions for SIEM rule ${ruleId}: ${e.message}`
     );
-
-    return {
-      resultedActions: [],
-      hasLegacyActions: false,
-      resultedReferences: [],
-    };
+    throw Boom.badRequest(
+      i18n.translate('xpack.alerting.rulesClient.validateLegacyActions.errorSummary', {
+        defaultMessage: 'Failed to migrate legacy actions for SIEM rule ${ruleId}: ${errorMessage}',
+        values: {
+          ruleId,
+          errorMessage: e.message,
+        },
+      })
+    );
   }
```

Comment:
if validation passes, but somehow rule update fails in rulesClient, we still can encounter data loss.
We not going to address it, but rather describe this case in documentation's known issues? Is that correct?

Maybe, would be worth to add logger.info the legacy actins were migrated for particular rule

Reply:
Yes, that s the plan. Tagging @nastasha-solomon here, to help us with writing docs. We will try to get these changes into 8.9 and we will need to add docs for 8.8.

Reply:
Thanks for the ping, @e40pud! To make sure I'm understanding you correctly, are you saying that the Detections Engine team will try to fix this bug in 8.9 and, in the meantime, it should be documented as a known issue in the 8.8 release notes? If so, would you mind filing a Security docs issue to track this doc task?

cc: @yctercero

Reply:
Yes, that is correct. We hope to merge this fix into `8.9.1` and add docs for `8.8`. I will create security docs ticket.

Reply:
@nastasha-solomon here is the ticket https://github.com/elastic/security-docs/issues/3572